File: languages/typescript/Title.ts
import { BaseElement } from './base_elements';

export interface Title extends BaseElement {
  // Nature
  privileges?: string | null;
  conditions?: string | null;
  createDate?: number | null;
  assignDate?: number | null;
  revokeDate?: number | null;
  hierarchy?: number | null;
  // Issue
  rights?: string | null;
  authorId?: string | null;
  // World
  characterId?: string | null;
  locationId?: string | null;
  objectId?: string | null;
  institutionId?: string | null;
  creatureId?: string | null;
  zoneId?: string | null;
  collectiveId?: string | null;
  constructId?: string | null;
}

File: languages/csharp_unity/Title.cs
using System;
[System.Serializable]
public class Title : BaseElement
{
    // Nature
    public string Privileges;
    public string Conditions;
    public int? CreateDate;
    public int? AssignDate;
    public int? RevokeDate;
    public int? Hierarchy;
    // Issue
    public string Rights;
    public string AuthorId;
    // World
    public string CharacterId;
    public string LocationId;
    public string ObjectId;
    public string InstitutionId;
    public string CreatureId;
    public string ZoneId;
    public string CollectiveId;
    public string ConstructId;
}
